packages/trackerless-network/src/logic/plumtree/PlumTreeManager.ts

```diff
@@ -0,0 +1,136 @@
+import { DhtAddress, ListeningRpcCommunicator, PeerDescriptor, toNodeId } from '@streamr/dht'
+import { 
+    MessageID,
+    PauseNeighborRequest,
+    ResumeNeighborRequest,
+    StreamMessage
+} from '../../../generated/packages/trackerless-network/protos/NetworkRpc'
+import { NodeList } from '../NodeList'
+import { PlumTreeRpcLocal } from './PlumTreeRpcLocal'
+import { PlumTreeRpcRemote } from './PlumTreeRpcRemote'
+import { ContentDeliveryRpcClient, PlumTreeRpcClient } from '../../../generated/packages/trackerless-network/protos/NetworkRpc.client'
+import EventEmitter from 'eventemitter3'
+import { Logger } from '@streamr/utils'
+import { ContentDeliveryRpcRemote } from '../ContentDeliveryRpcRemote'
+import { PausedNeighbors } from './PausedNeighbors'
+
+interface Options {
+    neighbors: NodeList
+    localPeerDescriptor: PeerDescriptor
+    rpcCommunicator: ListeningRpcCommunicator
+}
+
+const logger = new Logger(module)
+
+interface Events {
+    message: (msg: StreamMessage, previousNode: DhtAddress) => void
+}
+
+export class PlumTreeManager extends EventEmitter<Events> {
+    private neighbors: NodeList
+    private localPeerDescriptor: PeerDescriptor
+    private localPausedNeighbors: PausedNeighbors = new PausedNeighbors()
```

Comment:
add comments to clarify what are these localPausedNeighbors and remotePausedNeighbors